applications/FluidDynamicsApplication/custom_strategies/strategies/fs_strategy.h

```diff
@@ -112,107 +110,36 @@ class FSStrategy : public SolvingStrategy<TSparseSpace,TDenseSpace,TLinearSolver
-            pScheme.swap(Temp);
-        }
-        else
-        {
-            typename SchemeType::Pointer Temp = typename SchemeType::Pointer(new ResidualBasedIncrementalUpdateStaticScheme< TSparseSpace, TDenseSpace > ());
-            pScheme.swap(Temp);
-        }
-
-        //CONSTRUCTION OF VELOCITY
-        BuilderSolverTypePointer vel_build = BuilderSolverTypePointer(new ResidualBasedEliminationBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver > (pVelocityLinearSolver));
-        this->mpMomentumStrategy = typename BaseType::Pointer(new ResidualBasedLinearStrategy<TSparseSpace, TDenseSpace, TLinearSolver > (rModelPart, pScheme, pVelocityLinearSolver, vel_build, CalculateReactions, ReformDofAtEachIteration, CalculateNormDxFlag));
-        this->mpMomentumStrategy->SetEchoLevel( BaseType::GetEchoLevel() );
-
-        BuilderSolverTypePointer pressure_build = BuilderSolverTypePointer(
-                new ResidualBasedEliminationBuilderAndSolverComponentwise<TSparseSpace, TDenseSpace, TLinearSolver, Variable<double> >(pPressureLinearSolver, PRESSURE));
-
-        this->mpPressureStrategy = typename BaseType::Pointer(new ResidualBasedLinearStrategy<TSparseSpace, TDenseSpace, TLinearSolver > (rModelPart, pScheme, pPressureLinearSolver, pressure_build, CalculateReactions, ReformDofAtEachIteration, CalculateNormDxFlag));
-        this->mpPressureStrategy->SetEchoLevel( BaseType::GetEchoLevel() );
-
-        if (mUseSlipConditions)
-        {
-#pragma omp parallel
-            {
-                ModelPart::ConditionIterator CondBegin;
-                ModelPart::ConditionIterator CondEnd;
-                OpenMPUtils::PartitionedIterators(rModelPart.Conditions(),CondBegin,CondEnd);
-
-                for (ModelPart::ConditionIterator itCond = CondBegin; itCond != CondEnd; ++itCond)
-                {
-                    const bool is_slip = itCond->Is(SLIP);
-                    if (is_slip)
-                    {
-
-                        Condition::GeometryType& rGeom = itCond->GetGeometry();
-                        for (unsigned int i = 0; i < rGeom.PointsNumber(); ++i)
-                        {
-                            rGeom[i].SetLock();
-                            rGeom[i].Set(SLIP,is_slip);
-                            rGeom[i].UnSetLock();
-                        }
-                    }
-                }
-            }
-            rModelPart.GetCommunicator().SynchronizeOrNodalFlags(SLIP);
-        }
-
+        InitializeStrategy(rSolverConfig,PredictorCorrector);
+    }
 
-        KRATOS_CATCH("");
+    FSStrategy(
+        ModelPart& rModelPart,
+        SolverSettingsType& rSolverConfig,
+        bool PredictorCorrector,
+        bool CalculateReactionsFlag,
+        const Kratos::Variable<int>& PeriodicVar)
+        : BaseType(rModelPart,false)
+        , mCalculateReactionsFlag(CalculateReactionsFlag)
+        , mrPeriodicIdVar(PeriodicVar)
+    {
+        InitializeStrategy(rSolverConfig,PredictorCorrector);
```

Comment:
I see now that you have 4 constructors that essentially do the same

I recommend calling one constructor from the other (possible since C++11) like I did in `ModelPart`, this should reduce code duplication

up to you

Reply:
Considering that two constructors are deprecated, and hopefully we will remove them soon, I'd keep it as it is.